wandb/sdk/internal/handler.py

```diff
@@ -168,6 +175,86 @@ def _save_history(self, record: Record) -> None:
                 self._sampled_history.setdefault(k, sample.UniformSampleAccumulator())
                 self._sampled_history[k].add(v)
 
+    def _update_summary_metrics(
+        self,
+        s: wandb_internal_pb2.MetricSummary,
+        k: str,
+        v: "numbers.Real",
+        float_v: float,
+        goal_max: Optional[bool],
+    ) -> bool:
+        updated = False
+        best_key: Optional[str] = None
+        if s.best:
+            best_key = k + ".best"
+        if s.max or best_key and goal_max:
+            max_key = k + ".max"
+            old_max = self._metric_track.get(max_key)
+            if old_max is None or float_v > old_max:
+                self._metric_track[max_key] = float_v
+                if s.max:
+                    self._consolidated_summary[max_key] = v
+                    updated = True
+                if best_key:
+                    self._consolidated_summary[best_key] = v
+                    updated = True
+        # defaulting to minimize if goal is not supecified
+        if s.min or best_key and not goal_max:
+            min_key = k + ".min"
+            old_min = self._metric_track.get(min_key)
+            if old_min is None or float_v < old_min:
+                self._metric_track[min_key] = float_v
+                if s.min:
+                    self._consolidated_summary[min_key] = v
+                    updated = True
+                if best_key:
+                    self._consolidated_summary[best_key] = v
+                    updated = True
+        if s.mean:
+            tot_key = k + ".tot"
+            num_key = k + ".num"
+            avg_key = k + ".mean"
+            tot = self._metric_track.get(tot_key, 0)
+            num = self._metric_track.get(num_key, 0)
+            tot += float_v
+            num += 1
+            self._metric_track[tot_key] = tot
+            self._metric_track[num_key] = num
+            self._consolidated_summary[avg_key] = tot / num
```

Comment:
May not be needed but we might want to cast `tot` into a float.  There the unset case on line 217 which will be an integer and I imagine a user could pass in an int...